app/controllers/stages_controller.rb

```diff
@@ -133,9 +129,8 @@ def stage_params
       :update_github_pull_requests,
       :comment_on_zendesk_tickets,
       command_ids: [],
-      flowdock_flows_attributes: [:id, :name, :token, :_destroy],
-      new_relic_applications_attributes: [:id, :name, :_destroy]
-    )
+      new_relic_applications_attributes: [:id, :name, :_destroy],
```

Comment:
For IFrames we could use (postMessages)[https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage] to transfer information to the parent window every time an input would change. The parent form would be responsible to then send all the data to the controller. The controller can be the same, but that controller could send the plugin data to a plugin service object that knows how to store the configuration for all the plugins.

I was actually thinking that the plugins would be an easy way to get other people involved and contribute more. After all it seems that there is some curiosity on how to integrate this with other applications  (#256), (#52). This was part of my reasoning for having some kind of plugin templating DSL, to try to make it easier for other people to start and do something. Also, if we wanted to have a plugin that would support Mercurial instead of Git, we would not need a Rails engine. This is a simple example that we will have several classes of plugins with different requirements.

I am not against starting with simple rails engines. We should however look beyond that and try to prepare some kind of plugin framework, with some rules and conventions that can simplify plugin development and avoid boilerplate code. 
I agree that we should always be able to fallback to rails whenever it is really necessary.